compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -21,6 +22,61 @@ use super::{
 };
 use crate::ty;
 
+/// Functionality required for the bytes of an `Allocation`.
+pub trait AllocBytes:
+    Clone
+    + core::fmt::Debug
+    + Eq
+    + PartialEq
+    + PartialOrd
+    + Ord
+    + core::hash::Hash
+    + Deref<Target = [u8]>
+    + DerefMut<Target = [u8]>
+{
+    /// The address of the bytes.
+    fn expose_addr(&self) -> u64;
```

Comment:
I think it would be better to make this return a raw pointer, and leave exposing and things like that to the caller.

And in fact... why is this even needed? One can use the `deref` to get such a pointer, right?